vignettes/dev-guide/dg_table_hierarchy.Rmd

```diff
@@ -0,0 +1,95 @@
+
+isS4
+getclass - for class structure
+
+
+## Process and Methods
+
+We invite developers to use the provided examples to interactively explore the rtables hierarchy. The most helpful command is 'getClass' for a list of the slots associated with a class, in addition to related classes and their relative distances.
+
+## Representation of Information before generation
+
+
+## Table Representation
+"PredataAxisLayout" class is used to define the data subset instructions for tabulation. 2 subclasses (one for each axis): PredataColLayout, PredataRowLayout
+
+## Slots, Parent-Child Relationships
+
+## Content (summary row groups)
+
+Splits are core functionality for rtables as tabulation and calculations are often required on subsets of the data. 
+
+## Split Machinery
+```{r, message=FALSE}
+library(rtables)
+getClass("TreePos")
+```
+
+"TreePos" class contains split information as a list of the splits, split label values, and the subsets of the data that are generated by the split.
+
+AllSplit
+RootSplit
+MultiVarSplit
+VarStaticCutSplit
+CumulativeCutSplit
+VarDynCutSplit
+CompoundSplit
+VarLevWBaselineSplit
+
+
+The highest level of the table hierarchy belong to "TableTree". The code below identifies the slots associated with with this class. 
+```{r}
+getClass("TableTree")
+```
+
+As an S4 object, the slots can be accessed using "@" (similar to the use of "$" for list objects).
+You'll notice there are classes that fall under "Extends". The classes contained here have a relationship to the TableTree object and are "virtual" classes. To avoid the repetition of slots and carrying the same data (set of slots for example) that multiple classes may need, rtables extensively uses virtual classes. A virtual class cannot be instantiated, the purpose is for other classes to inherit information from it. 
+
+
+```{r}
+
+lyt <- basic_table(title = "big title")  %>%
+        split_rows_by("SEX", page_by = TRUE) %>%
+        analyze("AGE")
+
+tt <- build_table(lyt, DM)
+
+# Though we don't recommend using str for studying rtable objects, 
+# we do find it useful in this instance to visualize the parent/child relationships.
+str(tt, max.level=2)
+```
+
+## Tree Paths 
+
+Root to Leaves, are vectors of vectors 
+Tables are tree, nodes in the tree can have summaries associated with them. Tables are trees because of the nested structure. There is also the benefit of keeping and repeating necessary information when trying to paginate a table.
+
+Children of ElementaryTables are row objects. TableTree can have children that are either row objects or other table objects.
+
+
+#### TODO:
+Create Tree Diagram showing class hierarchy.
```

Comment:
@ayogasekaram Is this the only missing thing? I think we can eventually reiterate updates in the future, what do you think? ^^

Reply:
@ayogasekaram will you be adding this section in a later PR? If so, I think it's good to go for now.